sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -266,6 +298,19 @@ Scheduler::Scheduler() {
                      /*PropList=*/{}));
 }
 
+Scheduler::~Scheduler() {
+  // By specification there are several possible sync points: buffer
+  // destruction, wait() method of a queue or event. Stream doesn't introduce
+  // any synchronization point. It is guaranteed that stream is flushed and
+  // resources are released only if one of the listed sync points was used for
+  // the kernel. Otherwise resources for stream will not be released, issue a
+  // warning in this case.
+  if (StreamBuffersPool.size() > 0)
```

Comment:
Should access to this variable be guarded by a mutex?
Otherwise, one can workaround possible race-condition here with introducing an atomic variable which represents size of the map.